tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/markers/MarkerSupportViewTest.java

```diff
@@ -125,6 +130,59 @@ public void limitDisabled() throws Exception {
 		assertFalse(isLimitEnabled);
 	}
 
+	@Test
+	public void markerContentGeneratorExtensionLoaded() throws Exception {
+		MarkerSupportView view = (MarkerSupportView) PlatformUI.getWorkbench().getActiveWorkbenchWindow()
+				.getActivePage().showView(PROBLEM_VIEW_ID);
+
+		MarkerContentGenerator generator = getMarkerContentGenerator(view);
+		ContentGeneratorDescriptor descriptor = MarkerSupportRegistry.getInstance()
+				.getContentGenDescriptor(generator.getId());
+		assertNotNull(descriptor);
+
+		MarkerField[] allFields = descriptor.getAllFields();
+		boolean foundAdditionalField1 = false;
+		boolean foundRecursivelyAddedField2 = false;
+		for (MarkerField field : allFields) {
+			if (ProblemKeyMarkerField.class.equals(field.getClass())) {
```

Comment:
Maybe better to define some helper method that filters `allFields` with this `if` statement, then pack `field.getName()` to a result `List<String>`.

Then you can just do an `assertTrue("..., but found: " + names, names.contains("Problem Key"));`

This has the benefit of actually seeing what is in `allFields` w.r.t. the test; it would be easier to understand failures (especially sporadic ones). It would also reduce the length of the test and make it easier to read.